clippy_lints/src/use_self.rs

```diff
@@ -58,13 +58,15 @@ declare_clippy_lint! {
 pub struct UseSelf {
     msrv: Msrv,
     stack: Vec<StackItem>,
+    recursive_struct: bool,
 }
 
 impl UseSelf {
     pub fn new(conf: &'static Conf) -> Self {
         Self {
             msrv: conf.msrv,
             stack: Vec::new(),
+            recursive_struct: conf.recursive_self_in_struct,
```

Comment:
Even if it's longer to write, we need to write out the full configuration name for clarity.
If we didn't, a future maintainer could think that this boolean is `true` when encountering any recursive struct.

```suggestion
            recursive_self_in_struct: conf.recursive_self_in_struct,
```